### PR TITLE
Categorize tests in Allure per layer

### PR DESCRIPTION
**Pull request by bratashX**
- Add logic for extending allure reports.
- Add a new yaml file with mapping test to layer.

Example on Allure TestOps: https://hedgehog.testops.cloud/project/199/test-cases?treeId=0

### CODE REVIEW

File: tests/mapping_test_to_group.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+dhcp_relay.test_dhcp_relay#test_dhcp_relay_random_sport: L2
+dhcp_relay.test_dhcp_relay#test_dhcp_relay_start_with_uplinks_down: L2
+dhcp_relay.test_dhcp_relay#test_dhcp_relay_unicast_mac: L2
+dhcp_relay.test_dhcp_relay#test_interface_binding: L2
+dhcp_relay.test_dhcpv6_relay#test_dhcp_relay_after_link_flap: L2
+dhcp_relay.test_dhcpv6_relay#test_dhcp_relay_default: L2
+dhcp_relay.test_dhcpv6_relay#test_dhcp_relay_start_with_uplinks_down: L2
+dhcp_relay.test_dhcpv6_relay#test_dhcpv6_relay_counter: L2
+dhcp_relay.test_dhcpv6_relay#test_interface_binding: L2
+generic_config_updater.test_ntp#test_ntp_server_tc1_suite: L2
```
Comment (nazar-garmadiy) [inline tests/mapping_test_to_group.yaml]: probably NTP is "management", same for other NTP related tests

File: tests/mapping_test_to_group.yaml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+http.test_http_copy#test_http_copy: L2
```
Comment (nazar-garmadiy) [inline tests/mapping_test_to_group.yaml]: http_copy either L3 either "managenent"

File: tests/mapping_test_to_group.yaml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+pc.test_po_update#test_po_update: L2
+pc.test_po_update#test_po_update_io_no_loss: L2
+platform_tests.test_advanced_reboot#test_cancelled_fast_reboot: L2
+platform_tests.test_advanced_reboot#test_cancelled_warm_reboot: L2
+platform_tests.test_advanced_reboot#test_fast_reboot: L2
+platform_tests.test_advanced_reboot#test_warm_reboot: L2
+platform_tests.test_advanced_reboot#test_warm_reboot_mac_jump: L2
+platform_tests.test_advanced_reboot#test_warm_reboot_sad: L2
+platform_tests.test_cont_warm_reboot#test_continuous_reboot: L2
+radv.test_radv_ipv6_ra#test_radv_router_advertisement: L2
```
Comment (nazar-garmadiy) [inline tests/mapping_test_to_group.yaml]: RADVD (ipv6 router advertisement) - L3

File: tests/mapping_test_to_group.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+route.test_default_route#test_default_route_set_src: L2
+route.test_default_route#test_default_route_with_bgp_flap: L2
+route.test_route_flap#test_route_flap: L2
+route.test_route_perf#test_perf_add_remove_routes: L2
+route.test_static_route#test_static_route: L2
+route.test_static_route#test_static_route_ecmp: L2
+route.test_static_route#test_static_route_ecmp_ipv6: L2
+route.test_static_route#test_static_route_ipv6: L2
+scp.test_scp_copy#test_scp_copy: L2
+snmp.test_snmp_cpu#test_snmp_cpu: L2
```
Comment (nazar-garmadiy) [inline tests/mapping_test_to_group.yaml]: SNMP - "management"

File: tests/mapping_test_to_group.yaml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+snmp.test_snmp_loopback#test_snmp_loopback: L2
+snmp.test_snmp_pfc_counters#test_snmp_pfc_counters: L2
+snmp.test_snmp_queue#test_snmp_queues: L2
+ssh.test_ssh_ciphers#test_ssh_enc_ciphers: L2
+ssh.test_ssh_ciphers#test_ssh_kex: L2
+ssh.test_ssh_ciphers#test_ssh_macs: L2
+ssh.test_ssh_ciphers#test_ssh_protocol_version: L2
+ssh.test_ssh_limit#test_ssh_limits: L2
+ssh.test_ssh_stress#test_ssh_stress: L2
+tacacs.test_accounting#test_accounting_local_only: L2
```
Comment (nazar-garmadiy) [inline tests/mapping_test_to_group.yaml]: TACACS - "management"

File: tests/mapping_test_to_group.yaml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+tacacs.test_authorization#test_backward_compatibility_disable_authorization: L2
+tacacs.test_authorization#test_bypass_authorization: L2
+tacacs.test_jit_user#test_jit_user: L2
+tacacs.test_ro_user#test_ro_user: L2
+tacacs.test_ro_user#test_ro_user_allowed_command: L2
+tacacs.test_ro_user#test_ro_user_banned_command: L2
+tacacs.test_ro_user#test_ro_user_ipv6: L2
+tacacs.test_rw_user#test_rw_user: L2
+tacacs.test_rw_user#test_rw_user_ipv6: L2
+test_hedgehog_smoke#test_bfd_smoke: L2
```
Comment (nazar-garmadiy) [inline tests/mapping_test_to_group.yaml]: Let's consider "smoke" tests according to geature. E.g.:
database, swss, syncd - "management"
bgp - L3
lldp - L2

File: tests/mapping_test_to_group.yaml (hunk 1 above)
Comment (bratashX) [reply tests/mapping_test_to_group.yaml]: Updated

**Review by nazar-garmadiy — APPROVED** [review]

File: tests/hedgehog_test_runner.py [hunk 7]
```diff
@@ -4,8 +4,12 @@
 
 import paramiko
 import yaml
+import json
+
+from pathlib import Path
 
 FULL_REPORT_DIR_PATH = ""
+MAPPING_TEST_TO_GROUP_FILE = "./mapping_test_to_group.yaml"
```
Comment (andriidovhan) [inline tests/hedgehog_test_runner.py]: how about moving this file into `tests/hedgehog` directory ? 
just to keep all hedgehog related files in the same place

**Comment by andriidovhan** [comment]
as for me, would be great to have separate script for mapping test with corresponding `tag`.
I find it useful in case we have to run some list of tests for device which is not yet existed in `tests/hedgehog/env` directory.

if we decided to do it, we need to think: 
- what happens if this script runs twice for the same directory? it can be wrong directory or vpn is disconected, report_dir is partially updated and then we need to update the same report_dir

**Review by andriidovhan — APPROVED** [review]

File: tests/hedgehog_test_runner.py (hunk 7 above)
Comment (bratashX) [reply tests/hedgehog_test_runner.py]: Thanks, updated

**Pull request merged**